pkg/kmm/containers.go

```diff
@@ -46,18 +46,20 @@ func NewModLoaderContainerBuilder(modName string) *ModuleLoaderContainerBuilder
 }
 
 // WithModprobeSpec adds the specified Modprobe to the ModuleLoaderContainerBuilder.
-func (builder *ModuleLoaderContainerBuilder) WithModprobeSpec(
-	dirName, fwPath string, parameters, args, rawargs []string) *ModuleLoaderContainerBuilder {
+func (builder *ModuleLoaderContainerBuilder) WithModprobeSpec(dirName, fwPath,
+	moduleName string, parameters, args, rawargs, moduleLoadingOrder []string) *ModuleLoaderContainerBuilder {
```

Comment:
> Also, I think that by adding new params, the firmware test will need update.

I agree this is breaking the API, I would completely change the interface of this function to get a `modprobeSpec` instead of all its inner fields, this way you don't have to modify all the caller each time there is a new field to `modprobeSpec`.

Although this is not a "pure" dependency-injection, it is still more similar then what we see when creating new interfaces in golang.

For example, this function could look like:
```
func (builder *ModuleLoaderContainerBuilder) WithModprobeSpec(modprobeSpec v1beta1.ModprobeSpec) *ModuleLoaderContainerBuilder {
    if valid, _ := builder.validate(); !valid {
        return builder
    }

    glog.V(100).Infof(
        "Creating new ModuleLoaderContainerBuilder structure with modprobeSpec %v", modprobeSpec)

    builder.definition.Modprobe = modprobeSpec

    return builder
}
```

Which IMO is cleaner and don't depend on the implementation of  `modprobeSpec`

Reply:
> That one is being set, while the builder is being called.

You are right, I missed this one, I will fix it. Thanks.